stb_vorbis.c

```diff
@@ -86,6 +86,25 @@
 extern "C" {
 #endif
 
+typedef unsigned char  uint8;
+typedef   signed char   int8;
+typedef unsigned short uint16;
+typedef   signed short  int16;
+typedef unsigned int   uint32;
+typedef   signed int    int32;
+#if defined(__x86_64) || defined(__x86_64__) || defined(__amd64) || defined(_M_X64) \
```

Comment:
This is definitely wrong: `_M_X64` is an MSVC thing and long is always 32 bits in windows. The list can also be much simplified, to something like:
```c
#if defined(_LP64) || defined(__LP64__)
typedef unsigned long       uint64;
typedef   signed long        int64;
#elif defined(_WIN64)
typedef unsigned __int64    uint64;
typedef   signed __int64     int64;
#else /* assuming 32-bit arches: */
typedef unsigned long long  uint64;
typedef   signed long long   int64;
#endif
```

Reply:
Those `_M_ARM64` etc are all MSVC. All non-windows 64 bit arches follow the LP64 model so they define `_LP64` or `__LP64__` or both, hence the shortcut. Only windows is the weirdo who follows LLP64, hence the `_WIN32` special case.